src/init.cpp

```diff
@@ -1423,9 +1424,14 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
     if (GetBoolArg("-listenonion", DEFAULT_LISTEN_ONION))
         StartTorControl(threadGroup, scheduler);
 
+    Discover(threadGroup);
+
+    // Map ports with UPnP
+    MapPort(GetBoolArg("-upnp", DEFAULT_UPNP));
```

Comment:
nit: I think I would separate out the GetBoolArg put it into Options.

Then keep the calls to MapPort into Start/Stop.

seems a little more pragmatic & good for debugging (e.g., can dump a nodes options and know how the node was configured)